mdagent/tools/base_tools/analysis_tools/hydrogen_bonding_tools.py

```diff
@@ -0,0 +1,216 @@
+                plot_histogram(
+                    hb_counts,
+                    title=f"{method.capitalize()} Histogram",
+                    save_path=plot_save_path_hist,
+                )
+
+                plot_save_path_time_series = self.path_registry.get_mapped_path(
+                    f"{method}_time_series_plot.png",
+                )
+                plot_time_series(
+                    hb_counts,
+                    title=f"{method.capitalize()} Time Series",
+                    save_path=plot_save_path_time_series,
+                )
+                return """Succeeded. Analysis completed, results saved to file and plot
+                saved."""
+
+            else:
+                return """Failed. Path registry helps track
+                file locations and it is not set up. Please make sure it is set up
+                before running this tool."""
+
+        except Exception as e:
+            return f"Failed. {type(e).__name__}: {e}"
+
+    def top_file(self, traj_file: str) -> str:
+        top_file = os.path.join(os.path.dirname(traj_file), "topology.pdb")
+        return top_file
+
+    def save_results_to_file(self, results: dict, file_name: str) -> None:
+        with open(file_name, "w") as f:
+            json.dump(results, f)
+
+
+class KabschSander(BaseTool):
+    name = "kabsch_sander"
+    description = """Compute the hydrogen bond energy between each pair
+    of residues in every frame."""
+
+    path_registry: PathRegistry | None = None
+
+    def __init__(self, path_registry: PathRegistry):
+        super().__init__()
+        self.path_registry = path_registry
+
+    def _run(self, traj_file: str, top_file: str | None = None) -> str:
+        try:
+            if not top_file:
+                top_file = self.top_file(traj_file)
+            traj = load_single_traj(self.path_registry, top_file, traj_file)
+            if not traj:
+                return """Failed. Trajectory could not be loaded; unable to retrieve
+            data needed to find hydrogen bonds. This may be due to missing files,
+            corrupted files, or incorrect formatted file. Please check and try
+            again."""
+
+            result = md.kabsch_sander(traj)
+
+            if self.path_registry is not None:
+                result_dict = {
+                    "indices": [
+                        list(pair) for pair in result[0]
+                    ],  # Convert each tuple to a list
+                    "energies": result[1].tolist()
+                    if hasattr(result[1], "tolist")
+                    else list(result[1]),
+                }
+                self.save_results_to_file(result_dict, "kabsch_sander_results.json")
+                plot_save_path_time_series = self.path_registry.get_mapped_path(
+                    "kabsch_sander_time_series_plot.png",
+                )
+                plot_time_series(
+                    result[1],  # assuming result[1] contains the bond energies
```

Comment:
is this not always the case? why assumption?

Reply:
I wrote that because it is assuming if hydrogen bond is present. I can remove that.